queue_job/job.py

```diff
@@ -454,6 +465,37 @@ def __init__(self, func,
         self._eta = None
         self.eta = eta
         self.channel = channel
+        # Handle automatic.workflow.job, normally it is a normal model name.
+        # In practice only _validate_invoice_job is called through automatic.workflow.job
+        # This allows the user to select the field on account.invoice, instead of having
+        # to pick automatic.workflow.job, which doesn't have the fields of the invoice model.
+        if self.model_name == 'automatic.workflow.job':
+            model_name = 'account.invoice'
+            record_ids = self.env[model_name].sudo().browse(self.args)
+        else:
+            model_name = self.model_name
+            record_ids = self.recordset
+        sequence_rules = env['queue.sequence.rule'].sudo().search([('model_id.model', '=', model_name)])
+        if sequence_rules:
+            self.sequence_rule_ids = [(6, 0, sequence_rules.ids)]
+            # Change the following when implementing multiple rules per model
+            self.rule_name = sequence_rules[0].name
+            if len(sequence_rules) > 1:
+                _logger.warning('More than one sequence rule defined for %s',
+                                self.model_name)
+            if len(record_ids) > 1:
+                _logger.warning(
+                    'Applying sequence rule for job %s failed because it has multiple related '
+                    'records.', self.uuid
+                )
+            else:
+                for rule in sequence_rules:
+                    value = str(record_ids[rule.field_id.name])
+                    self.rule_value = value
```

Comment:
Only one rule is allowed per model: https://github.com/VanMoof/queue/blob/3151ffa11739660f4b56f10b6319e9b026714ee3/queue_job/models/queue_sequence_rule.py#L37-L41
I left some stubs to allow for multiple rules per model, as I had started with that approach. I mid-way decided not to implement multiple rules per model because it would make the solution more complex than what we need, especially considering not making the job queue any slower.